File: java/ql/test/library-tests/controlflow/dominance/Test2.java
import java.math.*;

class MyExn extends Throwable {}

public class Test2 {
	void f() throws Throwable {}

	void g(boolean b) throws Throwable {
		while (b) {
			if (b) {
			} else {
				try {
					f();
				} catch (MyExn e) {}
				;
			}
		}
	}
	
	void t(int x) {
		if (x < 0) {
			return;
		}
		while(x >= 0) {
			if (x > 10) {
				try {
					BigInteger n = new BigInteger( "wrong" );
				} catch ( NumberFormatException e ) { // unchecked exception
				}
			}
			x--;
		}
	}
}

File: java/ql/test/kotlin/library-tests/controlflow/dominance/Test2.kt
import java.math.*

class MyExn: Throwable() {}

public class Test2 {
    @Throws(Throwable::class)
	fun f() {}

    @Throws(Throwable::class)
	fun g(b: Boolean) {
		while (b) {
			if (b) {
			} else {
				try {
					f()
				} catch (e: MyExn) {}
				;
			}
		}
	}
	
	fun t(x: Int) {
		if (x < 0) {
			return
		}
        var y = x
		while(y >= 0) {
			if (y > 10) {
				try {
					val n: BigInteger = BigInteger( "wrong" );
				} catch (e: NumberFormatException) { // unchecked exception
				}
			}
			y--
		}
	}
}
